magento/ece-tools
# MCLOUD-6485: Add validator for braintree variable existence

### Description

Existance of `CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` variable in Magento >= 2.4.0 causes an error during `bin/magento app:config:import` command. In this task were added two validators which check that variable doesn't set.

https://jira.corp.magento.com/browse/MCLOUD-6485

- Added critical validator to the build phase which checks that variable CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL is absent in .mageto.app.yaml
- Added critical validator to the deploy phase to check that environment variable CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL doesn't exist

### Manual testing scenarios

1. Install Magento 2.4.0 from git https://wiki.corp.magento.com/pages/viewpage.action?spaceKey=MCP&title=ECE+Tools+-+Simple+Flow+to+Deploy+Magento+to+ECE+from+GitHub+Branch using `magento-cloud` 2.3.5 template
```
stage:
  global:
    DEPLOYED_MAGENTO_VERSION_FROM_GIT: '2.4.0'
    DEPLOY_FROM_GIT_OPTIONS:
      clear_magento_module_requirements: true
      repositories:
        ce:
          ref: 2.4.0-develop
          repo: https://<key>@github.com/magento/magento2ce.git
        ee:
          ref: 2.4.0-develop
          repo: https://<key>@github.com/magento/magento2ee.git
```
2. Push changes to the cloud and check that build phase failed with an error about redundant `CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` env variable in `.magento.app.yaml`
3. Add `CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` to env variables through UI interface: 
`env:CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` = Magento_Enterprise_Cloud_BT
4. Remove `CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` from `.magento.app.yaml` commit and push changes.
5. Check that the build phase was successful and deploy phase failed with an error about redundant `CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL` env variable 

### Release notes

For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud ECE-tools release notes](https://devdocs.magento.com/cloud/release-notes/ece-release-notes.html).

### Associated documentation updates

https://github.com/magento/devdocs/pull/7571

### Contribution checklist
